src/extractor/guidance/motorway_handler.cpp

```diff
@@ -354,7 +367,15 @@ Intersection MotorwayHandler::fromRamp(const EdgeID via_eid, Intersection inters
         BOOST_ASSERT(isMotorwayClass(intersection[1].eid, node_based_graph));
 
         intersection[1].instruction = getInstructionForObvious(
-            intersection.size(), via_eid, isThroughStreet(1, intersection), intersection[1]);
+            intersection.size(),
+            via_eid,
+            ::osrm::extractor::guidance::isThroughStreet(1,
+                                                         intersection,
+                                                         node_based_graph,
+                                                         node_data_container,
+                                                         name_table,
+                                                         street_name_suffix_table),
+            intersection[1]);
```

Comment:
Quick question regarding the `isThroughStreet` interface: at the moment we take an intersection and an index into the intersection. And the user has to make sure both correspond to each other.

Should we instead take an intersection and a connected road?

And maybe something for the future: should `ConnectedRoad` objects know the intersection they belong to?